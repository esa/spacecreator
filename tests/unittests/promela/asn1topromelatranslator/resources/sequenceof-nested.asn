SequencesOf DEFINITIONS ::= BEGIN

Type1 ::= SEQUENCE (SIZE(3)) OF INTEGER(0..2)

SimpleFixedSizeSequenceOf ::= SEQUENCE (SIZE(4)) OF Type1

END