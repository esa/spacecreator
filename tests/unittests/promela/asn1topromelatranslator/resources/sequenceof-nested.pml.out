inline INTEGER_generate_value(INTEGER_gv)
{
    if
    ::  atomic {
        select (INTEGER_gv : 0 .. 2);
    }
    fi;
}
inline Type1_generate_value(Type1_gv)
{
    int i;
    for(i : 0 .. 2)
    {
        INTEGER_generate_value(Type1_gv.data[i]);
    }
}
inline SimpleFixedSizeSequenceOf_generate_value(SimpleFixedSizeSequenceOf_gv)
{
    int i;
    for(i : 0 .. 3)
    {
        Type1_generate_value(SimpleFixedSizeSequenceOf_gv.data[i]);
    }
}
